src/rpdk/core/fragment/generator.py

```diff
@@ -75,11 +77,45 @@ def validate_fragments(self):
         self.__validate_no_transforms_present(raw_fragments)
         self.__validate_outputs(raw_fragments)
         self.__validate_mappings(raw_fragments)
+        lint_warnings = self.__validate_fragment_through_cfn_lint(raw_fragments)
+        if not lint_warnings:
+            LOG.warning("Module fragment is valid.")
+        else:
+            LOG.warning(
+                "Module fragment is probably valid, but there are "
+                "warnings/errors from cfn-lint "
+                "(https://github.com/aws-cloudformation/cfn-python-lint):"
+            )
+        for lint_warning in lint_warnings:
+            print(
+                "\t{} (from rule {})".format(lint_warning.message, lint_warning.rule),
+            )
 
     def __validate_outputs(self, raw_fragments):
         self.__validate_no_exports_present(raw_fragments)
         self.__validate_output_limit(raw_fragments)
 
+    @staticmethod
+    def __validate_fragment_through_cfn_lint(raw_fragment):
+        filename = "temporary_fragment.json"
+
+        with open(filename, "w") as outfile:
+            json.dump(raw_fragment, outfile, indent=4)
+
+        template = cfnlint.decode.cfn_json.load(filename)
+
+        # Initialize the ruleset to be applied (no overrules, no excludes)
+        # Runs Warning and Error rules
+        rules = cfnlint.core.get_rules([], [], [], [], False, [])
+
+        regions = ["us-east-1"]
```

Comment:
Should this region be hardcoded? What's the effect of this?

Reply:
`us-east-1` is the default for `cfn-lint` since it's close to a superset for AWS types

Reply:
Source: https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/config.py#L519

Reply:
I'm unsure whether there is a lot of benefit in having RPDK `validate` command receive a `region` parameter and passing this to cfn-lint. John also seemed to suggest above that we could make this whole validation optional. I don't want to make the `validate` CLI too complicated but i'm open to suggestions.

Reply:
Thanks for the clarification. Makes sense to me.